src/drivers/uavcan_v1/templates/_composite_type.j2

```diff
@@ -0,0 +1,115 @@
+{{ composite_type | definition_begin }}
+{
+{%- if T.fixed_port_id is not None %}
+	static constexpr CanardPortID PORT_ID = {{T.fixed_port_id}};
+{%- endif %}
+
+{%- set total_size = namespace(value=0) -%}
+{%- for field in composite_type.fields -%}
+    {% set total_size.value = total_size.value + field.data_type.bit_length_set|max %}
+{%- endfor %}
+	static constexpr size_t SIZE = {{total_size.value}};
+
+	static constexpr auto getDataTypeFullName() { return "{{T}}"; }
+
+{% for constant in composite_type.constants %}
+	static constexpr {{ constant.data_type | declaration }} {{ constant.name | id }} = {{ constant.value.native_value.numerator }} / {{ constant.value.native_value.denominator }};
+{%- endfor -%}
+
+{% if composite_type is UnionType %}
```

Comment:
You can either crib by Nunavut templates or just use them when my change is merged.

Reply:
I whipped up a fast example of what this code would look like in Nunavut:

https://github.com/thirtytwobits/nunavut/commit/020a4972692c22b3823e289a3467db8737019eb6

I need to talk to @pavel-kirienko about the canard dsdl stuff. We either need to move this all out into Nunavut as a support library or we need to accept that you have to use libcanard if you generate nunavut C w/serialization enabled.

Reply:
> We either need to move this all out into Nunavut as a support library or we need to accept that you have to use libcanard if you generate nunavut C w/serialization enabled.

I think we should reproduce the functionality of [`canard_dsdl.[ch]`](https://github.com/UAVCAN/libcanard/tree/master/libcanard) in the Nunavut C support library, keeping in mind also that Nunavut will be able to generate much more efficient serialization/deserialization code because it can leverage the field alignment information (for example, for most fields it will be able to use memcpy instead of bit copying because most fields are byte-aligned).

Since `canard_dsdl.[ch]` is an optional addition to libcanard, applications that leverage autogenerated code will simply omit that part of the library.